ee/codegen/src/generators/MetadataFilters.ts

```diff
@@ -0,0 +1,148 @@
+import { python } from "@fern-api/python-ast";
+import { AstNode } from "@fern-api/python-ast/core/AstNode";
+import { Writer } from "@fern-api/python-ast/core/Writer";
+
+import { VELLUM_CLIENT_MODULE_PATH } from "src/constants";
+import {
+  VellumLogicalCondition as VellumLogicalConditionType,
+  VellumLogicalConditionGroup as VellumLogicalConditionGroupType,
+  VellumLogicalExpression as VellumLogicalExpressionType,
+} from "src/types/vellum";
+
+export declare namespace MetadataFilters {
+  export interface Args {
+    metadata: VellumLogicalExpressionType;
+  }
+}
+
+export class MetadataFilters extends AstNode {
```

Comment:
We should name this to be more specific and ideally, even co-locate with the search node ast node.

```suggestion
export class SearchNodeMetadataFilters extends AstNode {
```